notebooks/Notebook3.md

```diff
@@ -18,17 +26,46 @@ exciting milestones, preliminary results, etc. Feel free to include images
 code, and any other resources that you think will help clearly convey your
 design process.
 
+This week I "completed" a draft program in my DSL, the card game Hearts, when I realized that there was 
+something severely wrong with my entire design. I had completely forgotten that Scala is not a 
+dynamically-typed language, and thus, I couldn't use classes in place of their subclasses, like I had in my 
+Python implementation of the language. The majority of the week was spent testing out different methods for 
+making my existing design work: the Dynamic trait, generic types, and even switching host languages. At the 
+very end of my work this week, I think I found an acceptable solution, which involves a HashMap of variables 
+and constants, plus liberal use of the class extension functionality of Scala, but I definitely want to talk 
+to other people in class about other ways I might go about tackling the problem. It definitely was 
+disappointing to go from feeling like my language was at a really good place to having such a severe setback, 
+but perhaps it will be for the best; the way I'm planning on doing things now won't require the user to have 
+to understand OOP, which would be really nice.
```

Comment:
Things like this do happen, especially when we are implementing in a new-to-us language. I hope you aren't too discouraged! As you mention, the difference between Python and Scala is illustrative for you personally. There might be some Scala features we can use to make things not so bad. Let's chat!

Reply:
I agree! Scala likes to mascaraed as python but a lot of its features are quite different  (I keep messing up about immutable lists), so your not alone. I would be suprised if the problems your facing don't have some built in tool to help.